feat: added notification support

corn-potage opened the pull request:
- added notification object const to define various notifications
- added IPC support to call methods from renderer
- added system check polling to add notification when free disk space is less than 40 GB, and when internet is down
- added empty notification view and button
- added notification service for sidebar
- added notification settings to preferences modal

jgresham commented on src/renderer/Generics/redesign/Toggle/Toggle.tsx:
```diff
@@ -46,12 +51,11 @@ export const Toggle = ({
           role: 'switch',
           id: 'flexSwitchCheckDefault',
-          defaultChecked: isChecked,
+          checked: isChecked,
```
I think defualtChecked may need to be set so that the component starts as a "controlled component". Not 100% sure. Check the warning in the console when opening the "Notifications preferences.." for the first time in app

```instrument.js:105 Warning: A component is changing an uncontrolled input to be controlled. This is likely caused by the value changing from undefined to a defined value, which should not happen. Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://reactjs.org/link/controlled-components```

jgresham reviewed (changes requested):
Awesome code and changes!

Nice job on the notifications "apis" over the IPC and using RTKQ!!

corn-potage commented:
@jgresham I fixed the issues, should be good now 👍  I also removed `enableBlinkFeatures: 'CSSColorSchemeUARendering'` because it seems like it's a security issue, and the scroll bars still change colors when switching light/dark modes.
